src/tier1/KeyValues.cpp

```diff
@@ -2607,7 +2607,15 @@ bool KeyValues::WriteAsBinary( CUtlBuffer &buffer )
 		{
 		case TYPE_NONE:
 			{
-				dat->m_pSub->WriteAsBinary( buffer );
+				if (dat->m_pSub)
```

Comment:
I'm not sure if I like this fix

Reply:
It's not great, but it *is* the correct fix. The previous version was depending on the function being able to handle a null `this`. (which is UB)